### DISCUSSION TITLE
where is initializationOptions ??

### DISCUSSION THREAD

**Discussion by Kan-Rup** [mfussenegger/nvim-jdtls#191] [Q&A]
In the documentation to set up with jdt.ls it says I have to extend initializationOptions. where is this file?
-----------------------------------------------------------------------------------------------------------------

java-debug installation

    Clone java-debug
    Navigate into the cloned repository (cd java-debug)
    Run ./mvnw clean install
    Extend the initializationOptions with which you start eclipse.jdt.ls as follows:

config['init_options'] = {
  bundles = {
    vim.fn.glob("path/to/java-debug/com.microsoft.java.debug.plugin/target/com.microsoft.java.debug.plugin-*.jar")
  };
}

----------------------------------------------------------------------------------
also it says extend the configuration for nvim-jdtls. there are many config files in the nvim-jdtls directory. what is the name of this file?
----------------------------------------------------------------------------------

nvim-dap setup

You also need to call require('jdtls').setup_dap() to have it register a java adapter.

To do that, extend the configuration for nvim-jdtls with:

config['on_attach'] = function(client, bufnr)
  -- With `hotcodereplace = 'auto' the debug adapter will try to apply code changes
  -- you make during a debug session immediately.
  -- Remove the option if you do not want that.
  require('jdtls').setup_dap({ hotcodereplace = 'auto' })
end

**Comment by mfussenegger** [comment]
The `config`  is referring to the `config` described in https://github.com/mfussenegger/nvim-jdtls#configuration

`init_options` are the `initializationOptions` as described in `:help vim.lsp.start_client`

**Reply by mfussenegger** [reply]
I also tried to clarify this in the readme:

https://github.com/mfussenegger/nvim-jdtls/pull/196
https://github.com/mfussenegger/nvim-jdtls/pull/197
